src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -654,25 +644,30 @@ SpinelNCPInstance::update_node_type(NodeType new_node_type)
 	}
 }
 
-
 void
 SpinelNCPInstance::update_link_local_address(struct in6_addr *addr)
```

Comment:
I agree and I was considering this, but  then decided against it for now... We have the link-local and mesh-local addresses as part of "wpanctl status" which I think is very helpful...

Within OpenThread we sort of treat the mesh-local and the thread link-local address (the one NCP calculates) as special case (e.g., user can add more link-local addresses but the NCP generated one is sort of kept/managed differently).